Bottom: 5b9a65d16f6a3f67bc98885320adfd136f12b16a
Top:    39ef7b27695684cfafb623d7e40a06be07ed9245
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-13 21:00:00 +0100

Add profiling to file load







---

diff --git a/src/file.c b/src/file.c
index 738188a..35bfb57 100644
--- a/src/file.c
+++ b/src/file.c
@@ -350,6 +350,10 @@ LoadPCB (char *Filename)
 {
   PCBTypePtr newPCB = CreateNewPCB (False);
   Boolean units_mm;
+  clock_t start, end;
+  double elapsed;
+
+  start = clock();
 
   /* new data isn't added to the undo list */
   if (!ParsePCB (newPCB, Filename))
@@ -380,6 +384,10 @@ LoadPCB (char *Filename)
 	  CreateDefaultFont ();
 	}
 
+  end = clock();
+  elapsed = ((double) (end - start)) / CLOCKS_PER_SEC;
+  printf ("Loading file %s took %f\n", Filename, elapsed);
+
       /* clear 'changed flag' */
       SetChangedFlag (False);
       PCB->Filename = MyStrdup (Filename, "LoadPCB()");
